### ISSUE TITLE
GraphQL Performance with and without Formie Plugin enabled

### ISSUE THREAD

**Issue by benediktblank** [verbb/formie#1563]
### Describe the bug

We have noticed some performance issues (general slowness) in our headless CraftCMS setup.
A relatively simple request just to get the ID of an entry can take between 1 and 2 seconds.

We were suspecting the plugins we have installed, one of them is Formie.
When I'm looking at a GQL query with the developer toolbar I can see that, with Formie it takes 650 mysql queries.
Without Formie active it takes around 400 mysql queries.
Also the time is reduced between 500ms and 1 second when Formie is disabled.

The thing is, my query does not even contain anything Formie related. I'm not requesting any Form data, just the ID of an Entry.
Why is there so much Formie involved (250 sql queries) when I'm not even requesting anything Formie related?

It seems that Formie makes "all" our GQL calls slower, not just the ones that involve Form handling.
It shouldn't require an additional 250 calls related to Formie things, if nothing in that query has anything to do with Formie.

It would be great if you can support us with this and give us feedback on what we've found out.
I can give you insights in the toolbar results if you want.

### Steps to reproduce

1.  Make a simple GQL query to request the ID of an entry
2.  Enable/Disable formie extension and compare runtimes

### Form settings

- Multi-page form: Yes or No
- Submission Method: Ajax or Page Reload
- Client-side Validation: Yes or No
- Custom Form Templates: Yes or No

### Craft CMS version

4.4.15

### Plugin version

2.0.36

### Multi-site?

Yes

**Comment by engram-design** [comment]
Hmm, I'll need to dive into the GQL internals for Craft, but my immediate thought is that the schema is being prepared for all forms+fields. This would be something run regardless of querying Formie content. But then, this should only happen once, not on every request.

**Comment by benediktblank** [comment]
> Hmm, I'll need to dive into the GQL internals for Craft, but my immediate thought is that the schema is being prepared for all forms+fields. This would be something run regardless of querying Formie content. But then, this should only happen once, not on ever request.

As far as I know, Craft does not cache that GQL schema as it's complex data that cannot be cached. I read that here:
https://github.com/craftcms/cms/issues/7222

**Comment by engram-design** [comment]
Right, that's an interesting read. I'll continue investigating what we can do.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.3](https://github.com/verbb/formie/releases/tag/4.0.0-beta.3).
